lib/Parse/ParseDecl.cpp

```diff
@@ -2829,6 +2828,32 @@ void Parser::ParseDeclarationSpecifiers(DeclSpec &DS,
       AttrsLastTime = true;
       continue;
 
+    case tok::kw__Checked:
+      // Checked C - it is parsed as checked array or checked function keyword
+      // checked array type, checked []
+      if (NextToken().is(tok::l_square)) {
+        if (!getLangOpts().CPlusPlus11 || !isCXX11AttributeSpecifier())
```

Comment:
In checked array, checked keyword is used before l_square (checked []).
In checked scope, checked keyword is used before l_brace (checked { ).
In checked function, checked keyword is used before another function specifier or type
To distinguish above two situations, I have used if-statement
In checked array case, it was same as before
In checked scope case, it is impossible case for this function declaration specifier
Otherwise, it is used as checked function specifier. In this case, I have set chekced function specifier in declaration

In checked array case, I will remove unnecessary code beneath the goto statement